src/lockfile/parse.js

```diff
@@ -314,9 +314,105 @@ export class Parser {
   }
 }
 
-export default function(str: string, fileLoc: string = 'lockfile'): Object {
-  str = stripBOM(str);
+const MERGE_CONFLICT_START = '<<<<<<<';
+const MERGE_CONFLICT_SEP = '=======';
+const MERGE_CONFLCIT_END = '>>>>>>>';
+
+/**
+ * Extract the two versions of the lockfile from a merge conflict.
+ */
+
+export function extractConflictVariants(str: string): Array<string> {
+  const variants: Array<Array<string>> = [[], []];
+  const lines = str.split(/\n/g);
+
+  while (lines.length) {
+    const line = lines.shift();
+    if (line.startsWith(MERGE_CONFLICT_START)) {
+      // get the first variant
+      let variantOne = [];
+      while (lines.length) {
+        const line = lines.shift();
+        if (line === MERGE_CONFLICT_SEP) {
+          break;
+        } else {
+          variantOne.push(line);
```

Comment:
can't you just push directly into `variants[0]`? or at least use `let variantOne = variants[0]` so that the concat becomes unneeded